lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1095,7 +1095,7 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
 
   Opts.EnableObjCInterop =
       Args.hasFlag(OPT_enable_objc_interop, OPT_disable_objc_interop,
-                   Target.isOSDarwin());
+                   Target.isOSDarwin() && !Opts.hasFeature(Feature::Embedded));
```

Comment:
Do we ever care about the non-darwin case explicitly enabling it with `enable-objc-interop`?